File: client/src/app/services/household.service.ts
// household.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { BaseCrudService } from './base-crud.service';
import { HouseholdBase as Household } from '../models/HouseholdBase';

@Injectable({
  providedIn: 'root'
})
export class HouseholdService extends BaseCrudService<Household> {
  constructor(protected override http: HttpClient) {
    super(http, 'households');
  }
}

File: api/services/households/household_service.py
"""Service for Household implementing CRUD operations."""
from sqlalchemy.orm import Session
from repositories.households.household_repository import HouseholdRepository
from schemas.households.household_schema import HouseholdCreate, HouseholdRead, HouseholdUpdate
from models import Household
from services.core.base_service import BaseService


class HouseholdService(BaseService[Household, HouseholdRead, HouseholdCreate, HouseholdUpdate]):
    """Service for Household domain logic."""

    def __init__(self, db: Session):
        super().__init__(
            db=db,
            model=Household,
            repository=HouseholdRepository(db),
            read_schema=HouseholdRead
        )
